rust/datafusion/src/catalog/information_schema.rs

```diff
@@ -221,3 +262,226 @@ impl InformationSchemaTablesBuilder {
+
+        // "Maximum length, in bytes, for binary data, character data,
+        // or text and image data."
+        let char_len: Option<u64> = match data_type {
+            Utf8 | Binary => Some(i32::MAX as u64),
+            LargeBinary | LargeUtf8 => Some(i64::MAX as u64),
+            _ => None,
+        };
+        self.character_octet_lengths
+            .append_option(char_len)
+            .unwrap();
+
+        // numeric_precision: "If data_type identifies a numeric type, this column
+        // contains the (declared or implicit) precision of the type
+        // for this column. The precision indicates the number of
+        // significant digits. It can be expressed in decimal (base
+        // 10) or binary (base 2) terms, as specified in the column
+        // numeric_precision_radix. For all other data types, this
+        // column is null."
+        //
+        // numeric_radix: If data_type identifies a numeric type, this
+        // column indicates in which base the values in the columns
+        // numeric_precision and numeric_scale are expressed. The
+        // value is either 2 or 10. For all other data types, this
+        // column is null.
+        //
+        // numeric_scale: If data_type identifies an exact numeric
+        // type, this column contains the (declared or implicit) scale
+        // of the type for this column. The scale indicates the number
+        // of significant digits to the right of the decimal point. It
+        // can be expressed in decimal (base 10) or binary (base 2)
+        // terms, as specified in the column
+        // numeric_precision_radix. For all other data types, this
+        // column is null.
+        let (numeric_precision, numeric_radix, numeric_scale) = match data_type {
+            Int8 | UInt8 => (Some(8), Some(2), None),
+            Int16 | UInt16 => (Some(16), Some(2), None),
+            Int32 | UInt32 => (Some(32), Some(2), None),
+            // From max value of 65504 as explained on
+            // https://en.wikipedia.org/wiki/Half-precision_floating-point_format#Exponent_encoding
+            Float16 => (Some(15), Some(2), None),
+            // Numbers from postgres `real` type
+            Float32 => (Some(24), Some(2), None),
+            // Numbers from postgres `double` type
+            Float64 => (Some(24), Some(2), None),
+            Decimal(precision, scale) => {
+                (Some(*precision as u64), Some(10), Some(*scale as u64))
+            }
+            _ => (None, None, None),
+        };
+
+        self.numeric_precisions
+            .append_option(numeric_precision)
+            .unwrap();
+        self.numeric_precision_radixes
+            .append_option(numeric_radix)
+            .unwrap();
+        self.numeric_scales.append_option(numeric_scale).unwrap();
+
+        self.datetime_precisions.append_option(None).unwrap();
+        self.interval_types.append_null().unwrap();
+    }
+
+    fn build(self) -> MemTable {
```

Comment:
it may be more idiomatic to use [`Into`](https://doc.rust-lang.org/std/convert/trait.Into.html), or `From`.

Reply:
Ok. Good idea

Reply:
In 98f702da90d18d8ae855a5eca6b0d1d6c1809551

Reply:
(which I now realize I put into https://github.com/apache/arrow/pull/9866 rather than this PR 🤦 ) -- but it should get in that PR hopefully